management/virtualmachine/client.go

```diff
@@ -19,23 +22,42 @@ const (
 	errParamNotSpecified = "Parameter %s is not specified."
 )
 
-//NewClient is used to instantiate a new VmClient from an Azure client
+//NewClient is used to instantiate a new VirtualMachineClient from an Azure client
 func NewClient(client management.Client) VirtualMachineClient {
 	return VirtualMachineClient{client: client}
 }
 
-func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName string) (management.OperationId, error) {
-	data, err := xml.Marshal(DeploymentRequest{
+// CreateDeployment creates a deployment and then creates a virtual machine
+// in the deployment based on the specified configuration.
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157194.aspx
+func (vm VirtualMachineClient) CreateDeployment(
+	role Role,
+	cloudServiceName string,
+	options *DeploymentOptions) (management.OperationId, error) {
```

Comment:
I don't think this should be accepting a pointer (and thus allow "nil" instead of zero value). We can allow fields inside `DeploymentOptions` to be pointers (which we do) and zero value will just work and we'll get rid of nil-check below.